File: domain/src/main/java/com/github/gmetal/domain/model/Session.java
package com.github.gmetal.domain.model;

public class Session {

    public static final int INVALID_SESSION_TYPE = -1;
    public static final int GUEST_SESSION_TYPE = 0;
    public static final int FULL_SESSION_TYPE = 1;

    private String mRequestToken;
    private String mSessionId;

    private String mRequestTokenExpirationDate;
    private String mSessionExpirationDate;

    private int mSessionType;

    public Session(final String sessionId, final int sessionType) {

        if (sessionId != null) {
            mSessionId = sessionId;
            mSessionType = sessionType;
        } else {
            mSessionType = INVALID_SESSION_TYPE;
        }
    }

    public boolean isInvalidSession() {

        return mSessionType == INVALID_SESSION_TYPE;
    }

    public boolean isGuestSession() {

        return mSessionType == GUEST_SESSION_TYPE;
    }

    public boolean isFullSession() {

        return mSessionType == FULL_SESSION_TYPE;
    }

    public int getSessionType() {

        return mSessionType;
    }

    public String getSessionId() {

        return mSessionId;
    }

    public String getRequestToken() {

        return mRequestToken;
    }

    public void updateRequestToken(final String requestToken) {

        mRequestToken = requestToken;
    }
    public void setSessionId(String sessionId) {

        mSessionId = sessionId;
        mSessionType = FULL_SESSION_TYPE;
    }
}

File: domain/src/main/java/com/github/gmetal/domain/model/Session.kt
package com.github.gmetal.domain.model

class Session(sessionId: String?, sessionType: Int) {

    var requestToken: String? = null
        private set
    private var mSessionId: String? = null

    private val mRequestTokenExpirationDate: String? = null
    private val mSessionExpirationDate: String? = null

    var sessionType: Int = 0
        private set

    val isInvalidSession: Boolean
        get() = sessionType == INVALID_SESSION_TYPE

    val isGuestSession: Boolean
        get() = sessionType == GUEST_SESSION_TYPE

    val isFullSession: Boolean
        get() = sessionType == FULL_SESSION_TYPE

    var sessionId: String?
        get() = mSessionId
        set(sessionId) {

            mSessionId = sessionId
            sessionType = FULL_SESSION_TYPE
        }

    init {

        if (sessionId != null) {
            mSessionId = sessionId
            this.sessionType = sessionType
        } else {
            this.sessionType = INVALID_SESSION_TYPE
        }
    }

    fun updateRequestToken(requestToken: String) {

        this.requestToken = requestToken
    }

    companion object {

        val INVALID_SESSION_TYPE = -1
        val GUEST_SESSION_TYPE = 0
        val FULL_SESSION_TYPE = 1
    }
}
